packages/ipfs-daemon/src/ipfs-daemon.ts

```diff
@@ -60,31 +80,40 @@ export class IpfsDaemon {
             ipfsSwarmTcpPort: props.ipfsSwarmTcpPort || Number(process.env.IPFS_SWARM_TCP_PORT) || 4011,
             ipfsSwarmWsPort: props.ipfsSwarmWsPort || Number(process.env.IPFS_SWARM_WS_PORT) || 4012,
 
-            ipfsEnableApi: props.ipfsEnableApi || (process.env.IPFS_ENABLE_API ? process.env.IPFS_ENABLE_API === 'true' : true),
+            ipfsEnableApi: props.ipfsEnableApi || fromBooleanInput(process.env.IPFS_ENABLE_API, true),
             ipfsApiPort: props.ipfsApiPort || Number(process.env.IPFS_API_PORT) || 5011,
 
-            ipfsEnableGateway: props.ipfsEnableGateway ?? (process.env.IPFS_ENABLE_GATEWAY ? process.env.IPFS_ENABLE_GATEWAY === 'true' : true),
+            ipfsEnableGateway: props.ipfsEnableGateway ?? fromBooleanInput(process.env.IPFS_ENABLE_GATEWAY, false),
             ipfsGatewayPort: props.ipfsGatewayPort || Number(process.env.IPFS_GATEWAY_PORT) || 9011,
-            ipfsDhtServerMode: props.ipfsDhtServerMode ?? (process.env.IPFS_DHT_SERVER_MODE === 'true'),
+            ipfsDhtServerMode: props.ipfsDhtServerMode ?? fromBooleanInput(process.env.IPFS_DHT_SERVER_MODE, false),
 
-            ipfsEnablePubsub: props.ipfsEnablePubsub ?? (process.env.IPFS_ENABLE_PUBSUB ? process.env.IPFS_ENABLE_PUBSUB === 'true' : true),
+            ipfsEnablePubsub: props.ipfsEnablePubsub ?? fromBooleanInput(process.env.IPFS_ENABLE_PUBSUB, true),
             ipfsPubsubTopics: props.ipfsPubsubTopics ?? (process.env.IPFS_PUBSUB_TOPICS ? process.env.IPFS_PUBSUB_TOPICS.split(' ') : []),
 
             ipfsBootstrap: ipfsBootstrap,
             ceramicNetwork: ceramicNetwork,
             useCentralizedPeerDiscovery: useCentralizedPeerDiscovery,
-            healthcheckEnabled: props.healthcheckEnabled ?? (process.env.HEALTHCHECK_ENABLED === 'true'),
+            healthcheckEnabled: props.healthcheckEnabled ?? fromBooleanInput(process.env.HEALTHCHECK_ENABLED, false),
             healthcheckPort: props.healthcheckPort || (process.env.HEALTHCHECK_PORT != null ? parseInt(process.env.HEALTHCHECK_PORT) : 8011),
             logger: props.logger ?? new DiagnosticsLogger(LogLevel.important, false),
         }
 
-        const repo = configuration.ipfsS3RepoEnabled ? createRepo({
+        const repo = createRepo({
             path: configuration.ipfsPath,
+            localPrefix: configuration.ipfsLocalPrefix,
```

Comment:
Can we rename to `localPathPrefix`? That way it is clear exactly what this option is for.

Reply:
Actually it would be nicer if we eliminate this options and just allow this to work

`IPFS_PATH=/my_dir_prefix/my_dir`

That should work with both s3 and local

Reply:
`IPFS_PATH` currently means two things at once:
1. Path to a local repository,
2. Bucket name on S3.

In case of purely local or purely s3 deployment, prefix is an unnecessary complication.

For mixed deployment, it might happen that `IPFS_PATH` points to a directory unavailable on a local machine. For example: `IPFS_PATH=/ipfs` in mixed setting makes use of `/ipfs` which might be unavailable on [certain](https://github.com/coreos/fedora-coreos-tracker/issues/270) [operating](https://cloud.google.com/container-optimized-os/docs/concepts/security#filesystem) [systems](https://support.apple.com/en-gb/HT210650).

Reply:
So, I am up for a rename, but I argue it has to stay with us.